src/main/java/mate/academy/service/impl/AuthenticationServiceImpl.java

```diff
@@ -0,0 +1,45 @@
+package mate.academy.service.impl;
+
+import java.util.Optional;
+import mate.academy.exception.AuthenticationException;
+import mate.academy.exception.RegistrationException;
+import mate.academy.lib.Inject;
+import mate.academy.lib.Service;
+import mate.academy.model.User;
+import mate.academy.service.AuthenticationService;
+import mate.academy.service.UserService;
+import mate.academy.util.HashPasswordUtil;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    private UserService userService;
+
+    @Override
+    public User login(String email, String password) throws AuthenticationException {
+        Optional<User> userOptional = userService.findByEmail(email);
+        if (userOptional.isPresent()) {
+            User user = userOptional.get();
+            if (user.getPassword()
+                    .equals(HashPasswordUtil.hashPassword(password, user.getSalt()))) {
+                return user;
+            }
+        }
+        throw new AuthenticationException("Invalid email or password");
+    }
+
+    @Override
+    public User register(String email, String password) throws RegistrationException {
+        if (email.isBlank()) {
+            throw new RegistrationException("Email should be not null or blank");
+        }
+        if (password.isBlank()) {
+            throw new RegistrationException("Password should be not null or blank");
+        }
```

Comment:
Not sure if these checks are needed here or at the controller level

Reply:
in the future such format checks will be done by declaring field constraints on the dto class, so there will be no need in manually check for null or similar data format checks but for now, it is fine to check it here